lib/skate/settings/user.ex

```diff
@@ -54,22 +55,9 @@ defmodule Skate.Settings.User do
             on_conflict: {:replace, [:email]}
           )
 
-        # existing user has same username - no update needed
-        user_matching_email.username == username ->
-          user_matching_email
-
-        # username format has changed. Create a new record for this user without an associated email
-        # This record will be used only temporarily until user settings are universally fetched by email.
-        user_matching_email.username != username ->
-          Skate.Repo.insert!(
-            DbUser.changeset(%DbUser{}, %{
-              username: username
-            }),
-            returning: true,
-            conflict_target: [:username],
-            # update a row with no effect so the returning works
-            on_conflict: {:replace, [:username]}
-          )
+        existing_user_with_email ->
+          # If there is an existing user with this email, return it without modifying the username
+          existing_user_with_email
```

Comment:
> without modifying the username

This is due to a limitation of the old case "username format has changed. Create a new record for this user without an associated email". For example, now for some users there are two records in the db:
`%{email: "kb@fake_email.com", username: "old_format\kb"}`
`%{email: nil, username: "new_format\kb"}`

user "new_format\kb" still has email `"kb@fake_email.com", but we didn't save it in order to enforce the unique constraint on email. So when "new_format\kb" signs in, this will find user `%{email: "kb@fake_email.com", username: "old_format\kb"}` and return it. We cannot update the username to "new_format\kb" without violating the unique constraint. 

Once all the old records without email addresses are removed from the DB, we can change this clause to update the username (if the username field still exists and is in use).